shell/translations.php

```diff
@@ -0,0 +1,232 @@
+                foreach ($nodes as $node) {
+                    // Which children should we translate?
+                    $translate_children = array_map('trim', explode(' ', $node['translate']));
+                    foreach ($node->children() as $child) {
+                        if (in_array($child->getName(), $translate_children)) {
+                            $matches[] = $child->__toString();
+                        }
+                    }
+                }
+            }
+
+            $matches = array_filter(array_unique($matches));
+            if (count($matches)) {
+                $map[$file] = $matches;
+            }
+        }
+        return $map;
+    }
+
+
+    /**
+     * Run script
+     *
+     */
+    public function run()
+    {
+        if ($this->getArg('missing')) {
+
+            $defined_file_map = $this->getDefinedStrings();
+            $used_file_map = $this->getUsedStrings();
+
+            $defined_flat = array_unique(array_merge(...array_values($defined_file_map)));
+            $used_flat = array_unique(array_merge(...array_values($used_file_map)));
+
+            if ($this->getArg('verbose')) {
+                foreach ($used_file_map as $file => $used) {
+                    $missing = array_diff($used, $defined_flat);
+                    if (count($missing)) {
+                        echo "$file\n    " . implode("\n    ", $missing) . "\n\n";
+                    }
+                }
+            } else {
+                $missing = array_diff($used_flat, $defined_flat);
+                sort($missing);
+                echo implode("\n", $missing) . "\n";
+            }
+
+        } else if ($this->getArg('unused')) {
+
+            $defined_file_map = $this->getDefinedStrings();
+            $used_file_map = $this->getUsedStrings();
+
+            $defined_flat = array_unique(array_merge(...array_values($defined_file_map)));
+            $used_flat = array_unique(array_merge(...array_values($used_file_map)));
+
+            if ($this->_stdin) {
+                echo "stdin file list cannot be used with the 'unused' mode.\n";
+                exit;
+            }
+            if ($this->getArg('verbose')) {
+                foreach ($defined_file_map as $file => $defined) {
+                    $unused = array_diff($defined, $used_flat);
+                    if (count($unused)) {
+                        echo "$file\n    " . implode("\n    ", $unused) . "\n\n";
+                    }
+                }
+            } else {
+                $unused = array_diff($defined_flat, $used_flat);
+                sort($unused);
+                echo implode("\n", $unused) . "\n";
+            }
+
+        } else {
+            echo $this->usageHelp();
+        }
+    }
+
+    /**
+     * Retrieve Usage Help Message
+     *
+     */
+    public function usageHelp()
+    {
+        return <<<USAGE
+Usage:  php -f translations.php -- [options]
+        php -f translations.php -- missing --verbose
```

Comment:
would it be possibile to have "--missing" and "--unused" instead of "-- missing"? would you like it that way?